### ISSUE TITLE
Only 50 last tasks shown for workflows with many tasks

### ISSUE THREAD

**Issue by johandahlberg** [StackStorm/st2web#312]
We have a workflow which generates tasks in mistral using the `with-items` syntax and since this needs to be carried out for a large number of item we get a very large workflow. Looking at this workflow in the webui I noticed that it appears that when a workflow has to many tasks only the last 50 are shown in the webui. While all the information is still available via the cli it might make troubleshooting these very large workflow more difficult.

**Comment by enykeev** [comment]
Thanks for the report and sorry for the late reply.

Indeed, we're limiting a single query to `/executions` endpoint to 50 entries to keep UI responsiveness at bay and also prevent browser JSON parser from hanging (which is sadly the case even at this day and age).

How big of a workflow execution are we talking about in your case? Is is just a few executions that goes missing or are we talking about hundreds?

**Comment by johandahlberg** [comment]
Sorry for taking so long to come back to you on this - I've been on vacation. We currently have a workflow which generates about 500 tasks... This might not be the best way to go about things, but except for not showing up in the web-ui, it appears to be working.

I do realize that this kind of workflow might be kind of niche, so feel free to close this issue if you think that it is out of scope.

**Comment by enykeev** [comment]
Yes, 500 tasks per workflow execution invalidates most of our UX assumptions on the spot. Loading them all in one go is out of the question, even simple "show more" approach to pagination would quickly kill the browser with sheer amount of DOM nodes required.

I wonder, how it works for you in CLI? How long it takes to `st2 execution get` such a workflow? Are you using grep to sort through them or just doing it by eye?

All in all, it seem to me that you need an entirely different set of tools to work with such a huge workflows. At the moment we're starting to explore things we can improve for v2 API and this issues seem like an interesting direction we should look into.

As for an immediate action, we need to at least let user know there's more tasks in this workflow than UI is showing. I'll look into it.

**Comment by johandahlberg** [comment]
It's not that fast, but acceptable in the CLI. Takes about 7 s to display. And we do use grep to sort through them for troubleshooting.

Sounds like a good idea to add some type of notice that not all tasks are showing - left me a bit confused until I figured out only 50 of them were showing... :)

**Comment by christopherjbaker** [comment]
With the expanded filters (#133), it could include a way to filter by parent id (defaulting to the current `'null'`).

**Comment by jbrownsc** [comment]
I'm hitting this same limitation using Stack Storm 2.6.0.  We have 75 actions that get executed, some are in retry loops, and only the last 50 are displayed under the parent workflow in UI.
